internal/zinc-core/src/main/scala/sbt/internal/inc/Stamp.scala

```diff
@@ -141,8 +127,25 @@ object Stamper {
     catch { case i: IOException => EmptyStamp }
   }
 
-  val forHash = (toStamp: File) => tryStamp(Hash.ofFile(toStamp))
-  val forLastModified = (toStamp: File) => tryStamp(new LastModified(toStamp.lastModified()))
+  private final val xxHash = LongHashFunction.xx()
+  private final val BufferSize = 8192
+  final val forHash: (File => Stamp) = (toStamp: File) => {
+    if (!toStamp.exists() || toStamp.isDirectory) EmptyStamp
+    else {
+      sbt.io.Using.fileInputStream(toStamp) { (is: InputStream) =>
+        val acc = new Array[Long](2)
+        val buffer = new Array[Byte](BufferSize)
+        while (is.read(buffer) >= 0) {
+          val checksumChunk = xxHash.hashBytes(buffer)
```

Comment:
@stuhood I'm looking into a way of mixing hashes safely. 
Note that Guava does not support xxHash. I like our current dependency because is high-quality and self-contained, and pretty light.